app/models/node.rb

```diff
@@ -0,0 +1,11 @@
+class Node < ActiveRecord::Base
+  belongs_to :node
+  has_many :nodes
+  belongs_to :tree
+
+  # A child must point to its parent, hence 'node_id: self.id'
+  # for whatever reason, active record did not do this automatically
+  def create_node (args = {})
```

Comment:
formatting error. `def create_node (args = {})` should be `def create_node(args = {})`